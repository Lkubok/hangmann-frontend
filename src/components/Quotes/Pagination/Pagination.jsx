import React, { PureComponent } from "react";
import ReactPaginate from "react-paginate";
import { connect } from "react-redux";
import * as selectors from "../../../reducers/selectors";
import { changeActualPage } from "../../../actions/quoteActions";
import "./Pagination.scss";

export class Pagination extends PureComponent {
  handlePageClick = e => this.props.changePage(e.selected);

  render() {
    const { filterQuery, pagesCount, actualPage, pageLimit } = this.props;
    return (
      <ReactPaginate
        key={`${filterQuery}+${pageLimit}`}
        previousLabel="previous"
        nextLabel="next"
        breakLabel="..."
        breakClassName="break-me"
        pageCount={pagesCount}
        marginPagesDisplayed={2}
        pageRangeDisplayed={5}
        onPageChange={this.handlePageClick}
        containerClassName="pagination"
        subContainerClassName="pages pagination"
        activeClassName="page-active"
        initialPage={actualPage}
      />
    );
  }
}

const mapStateToPops = (state, ownProps) => ({
  pagesCount: selectors.pagesCount(state),
  actualPage: selectors.getActualPage(state),
  filterQuery: selectors.getFilterQuery(state),
  pageLimit: selectors.getPageLimit(state)
});

const mapDispatchToProps = dispatch => ({
  changePage: arg => dispatch(changeActualPage(arg))
});

export default connect(
  mapStateToPops,
  mapDispatchToProps
)(Pagination);
