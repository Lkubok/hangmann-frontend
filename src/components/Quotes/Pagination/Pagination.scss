@import "../../../scss/mixins.scss";
@import "../../../scss/variables.scss";

.pagination {
  flex-basis: 50%;
}

.pagination ul {
  display: inline-block;
  padding-left: 0px;
  padding-right: 15px;
}

.pagination li {
  display: inline-block;
  a {
    padding: 5px;
    transition: all 0.4s ease-in-out;
    &:hover {
      cursor: pointer;
      background-color: $primary-color;
      color: $dark-color;
      border-radius: 5px;
    }
  }
}

.page-active {
  color: $primary-color;
}
