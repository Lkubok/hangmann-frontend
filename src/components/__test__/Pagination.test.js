import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Pagination } from "../Quotes/Pagination/Pagination";
Enzyme.configure({ adapter: new Adapter() });

describe("Pagination component", () => {
  beforeEach(() => {});
  it("Should match snapshot", () => {
    const wrapper = shallow(<Pagination />);
    expect(wrapper).toMatchSnapshot();
  });
});
