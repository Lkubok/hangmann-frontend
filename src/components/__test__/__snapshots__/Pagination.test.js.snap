// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component Should match snapshot 1`] = `
<PaginationBoxView
  activeClassName="page-active"
  breakClassName="break-me"
  breakLabel="..."
  containerClassName="pagination"
  disableInitialCallback={false}
  disabledClassName="disabled"
  key="undefined+undefined"
  marginPagesDisplayed={2}
  nextClassName="next"
  nextLabel="next"
  onPageChange={[Function]}
  pageCount={10}
  pageRangeDisplayed={5}
  previousClassName="previous"
  previousLabel="previous"
  subContainerClassName="pages pagination"
/>
`;
